metpy/calc/basic.py

```diff
@@ -773,6 +773,163 @@ def smooth_n_point(scalar_grid, n=5, passes=1):
     return smooth_grid
 
 
+@exporter.export
+@preprocess_xarray
+def altimeter_to_station_pressure(altimeter_value, height):
+    r"""Convert the altimeter measurement to station pressure.
+
+    This function is useful for working with METARs since most will provide
+    altimeter values, but not sea-level pressure or station pressure.
+    The following definitions of altimeter setting and station pressure
+    are taken from the Federal Meteorology Handbook (2013) p. 11-1.
+    Altimeter setting is the pressure value to which an aircraft altimeter scale
+    is set so that it will indicate the altitude above mean sea-level of an aircraft
+    on the ground at the location for which the value is determined. It assumes a
+    standard atmosphere. Station pressure is the atmospheric pressure at the
+    designated station elevation. Finding the station pressure can be helpful for
+    calculating sea-level pressure or other parameters.
+
+    Parameters
+    ----------
+    altimeter_value : `pint.Quantity`
+            The altimeter setting value as defined by the METAR or other observation,
+            which can be measured in either inches of mercury (in. Hg) or millibars (mb)
+    height: `pint.Quantity`
+            Elevation of the station measuring pressure.
+
+    Returns
+    --------
+    station_pressure: `pint.Quantity`
+            The station pressure in hPa, which can be used to calculate sea-level pressure
+
+    See Also
+    ---------
+    altimeter_to_sea_level_pressure
+
+    Notes
+    -------
+    This function is implemented using the following equations from the
+    Smithsonian Handbook (1951) p. 269
```

Comment:
Can you add the Smithsonian Handbook to `references.rst` and cite from there like we do elsewhere in the docs? It'd probably also be good to put a copy in `docs/_static`. I have the PDF if you need.